pointer.go

```diff
@@ -142,6 +144,17 @@ func (p *JsonPointer) implementation(i *implStruct) {
 				return
 			}
 
+		case bson.D:
+			decodedToken := decodeReferenceToken(token)
```

Comment:
tiny nit: you could put the `v.Map()` on this line and assign it to a variable, so you don't need to pay the cost of calling it twice on both lines 149 and 150 (it gets slower as the object gets bigger).